packages/nextjs/src/app-router/client/ClerkProvider.tsx

```diff
@@ -23,14 +24,13 @@ export const ClientClerkProvider = (props: NextClerkProviderProps) => {
 
   useSafeLayoutEffect(() => {
     window.__unstable__onBeforeSetActive = () => {
-      if (__unstable_invokeMiddlewareOnAuthStateChange) {
-        router.refresh();
-        router.push(window.location.href);
-      }
+      return invalidateCacheAction();
```

Comment:
Wouldn't we want to invalidate the router cache in `onAfterSetActive`? Why in before?

Reply:
We want to invalidate the router cache before a navigation occurs, otherwise the navigation might not fire atll . Currently, the order is: setActive > onBeforeSetActive > beforeEmit() > onAfterSetActive